apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/admin/api/client/BftScanConnection.scala

```diff
@@ -1015,16 +1026,100 @@ object BftScanConnection {
+      override val getScans: BftScanConnection => Future[Seq[DsoScan]],
+      override val scansRefreshInterval: NonNegativeFiniteDuration,
+      override val retryProvider: RetryProvider,
+      override val loggerFactory: NamedLoggerFactory,
+  )(implicit override val ec: ExecutionContext)
+      extends Bft {
+
+    override protected def filterScans(allScans: Seq[DsoScan]): Seq[DsoScan] = allScans
+
+    override protected def validateState(state: BftState): Unit = {
+      val defaultCallConfig = BftCallConfig.default(state.scanConnections)
+
+      if (!defaultCallConfig.enoughAvailableScans) {
+        throw io.grpc.Status.FAILED_PRECONDITION
+          .withDescription(
+            s"There are not enough Scans to satisfy f=${state.scanConnections.f}. Will be retried. State: $state.scanConnections"
+          )
+          .asRuntimeException()
+      }
+    }
+
+    override def scanConnections: ScanConnections = {
+      val connections = currentConnectionsState
+      connections.copy(totalInSubset = None, threshold = None)
+    }
+  }
+
+  class BftCustom(
+      val trustedSvs: NonEmptyList[String],
+      val threshold: Option[Int],
+      override val initialScanConnections: Seq[SingleScanConnection],
+      override val initialFailedConnections: Map[Uri, Throwable],
+      override val connectionBuilder: Uri => Future[SingleScanConnection],
+      override val getScans: BftScanConnection => Future[Seq[DsoScan]],
+      override val scansRefreshInterval: NonNegativeFiniteDuration,
+      override val retryProvider: RetryProvider,
+      override val loggerFactory: NamedLoggerFactory,
+  )(implicit override val ec: ExecutionContext, tc: TraceContext)
+      extends Bft {
+
+    private val trustedSvsSet = trustedSvs.toList.toSet
+
+    override protected def filterScans(allScans: Seq[DsoScan]): Seq[DsoScan] = {
+      val targetScans = allScans.filter(scan => trustedSvsSet.contains(scan.svName))
+
+      logger.info(s"Discovered the following trusted scans from the network: ${targetScans
+          .map(s => s"Name=${s.svName}, URL=${s.publicUrl}")
+          .mkString("; ")}")
+      targetScans
+    }
+
+    override protected def validateState(state: BftState): Unit = {
+      val successfulCount = state.openConnections.size
+      val requiredCount = threshold.getOrElse((trustedSvs.size / 3) + 1)
+
+      if (successfulCount < requiredCount) {
+        val successfulNames = state.openConnections.values.map(_._2).toSet
+        val failedNames = trustedSvs.toList.filterNot(successfulNames.contains).mkString(", ")
+
+        throw Status.UNAVAILABLE
+          .withDescription(
+            s"Failed to connect to required number of trusted scans. " +
+              s"Required: $requiredCount, Connected: $successfulNames. Scan apps failed or missing from network: $failedNames."
+          )
+          .asRuntimeException()
+      } else {
+        logger.info(s"created threshold number of scan connections.")
+      }
```

Comment:
this seems to be pretty similar so i would generalize it and keep in the specific subclasses just the threshold calculation